mutiny-core/src/nostr/mod.rs

```diff
@@ -1172,6 +1197,242 @@ impl<S: MutinyStorage> NostrManager<S> {
         Ok(event_id)
     }
 
+    /// Creates a recommendation event for a federation
+    pub async fn recommend_federation(
+        &self,
+        invite_code: &InviteCode,
+    ) -> Result<EventId, MutinyError> {
+        let kind = Kind::from(18173);
+        // first pull down our current recommendation event, we need to do this
+        // to make sure we don't overwrite any other recommendations
+        let filter = Filter::new().kind(kind).author(self.public_key).limit(1);
+        // fetch events
+        let events = self
+            .client
+            .get_events_of(vec![filter], Some(Duration::from_secs(5)))
+            .await?;
+
+        let current = events
+            .into_iter()
+            .find_map(|e| if e.kind == kind { Some(e) } else { None });
+
+        // tag the federation invite code
+        let invite_code_tag = Tag::Generic(
+            TagKind::U,
+            vec![invite_code.to_string(), "fedimint".to_string()],
+        );
+
+        // todo tag the federation announcement event, to do so we need to have the pubkey of the federation
+
+        // create the new recommendation event by either creating a new one or modifying the current one
+        let builder = match current {
+            Some(e) => {
+                let mut tags = e.tags.clone();
+
+                let mut modified = false;
+
+                if !tags.iter().any(|t| t == &invite_code_tag) {
+                    tags.push(invite_code_tag);
+                    modified = true;
+                }
+
+                // if we already recommended this federation, return the event id
+                // no need to send another recommendation
+                if !modified {
+                    return Ok(e.id);
+                }
+
+                EventBuilder::new(kind, e.content.clone(), tags)
+            }
+            None => EventBuilder::new(kind, "", [invite_code_tag]),
+        };
+
+        // send the event
+        Ok(self.client.send_event_builder(builder).await?)
+    }
+
+    /// Queries our relays for federation announcements
+    pub async fn discover_federations(&self) -> Result<Vec<NostrDiscoveredFedimint>, MutinyError> {
+        // get contacts by npub
+        let mut npubs: HashMap<nostr::PublicKey, Contact> = self
+            .storage
+            .get_contacts()?
+            .into_iter()
+            .filter_map(|(_, c)| c.npub.map(|npub| (npub, c)))
+            .collect();
+
+        const NUM_TRUSTED_USERS: u32 = 500;
```

Comment:
what if this was higher to like 1k? 

Also, why `const` right here? Stick it towards the top.